paasta_tools/utils.py

```diff
@@ -1016,6 +1016,26 @@ def get_paasta_branch(cluster, instance):
     return get_paasta_branch_from_deploy_group('%s.%s' % (cluster, instance))
 
 
+def format_timestamp(dt=None):
+    if dt is None:
+        dt = datetime.datetime.utcnow()
+    return dt.strftime('%Y%m%dT%H%M%S')
+
+
+def get_paasta_tag_from_deploy_group(identifier, desired_state):
```

Comment:
There is quite a bit of refactoring that should get done. I was going to hold off until I finish the second half of this change before doing it.